### DISCUSSION TITLE
Reset Firewall State Table Function

### DISCUSSION THREAD

**Discussion by SleepyNinja0o** [jaredhendrickson13/pfsense-api#538] [Ideas]
I am currently working on a project that implements a strict Firewall lockdown and state table reset in the case of an active intrusion. I've noticied the API only allows deleting Firewall states individually. Would it be difficult to implement an entire state table reset? pfSense already has an endpoint for this function at /diag_resetstate.php.

**Comment by jaredhendrickson13** [comment]
This is implemented by #494 and is present in the `next_minor` branch. You'll be able to kill firewall states using a query, which also allows you to kill all states. It will be included in v2.1.0 which should release before the end of the month.

**Reply by SleepyNinja0o** [reply]
Awesome, great to hear! Thanks!
